docs/extras/building_vtk.rst

```diff
@@ -2,78 +2,256 @@
+
+.. code-block:: bash
+
+    #!/bin/bash
+    # builds python wheels on docker container and tests installation
+
+    set -e -x
+
+    # build based on python version from args
+    PYTHON_VERSION="$1"
+    case $PYTHON_VERSION in
+    2.7)
+      PYBIN="/opt/python/cp27-cp27m/bin/python"
+      ;;
+    3.5)
+      PYBIN="/opt/python/cp35-cp35m/bin/python"
+      ;;
+    3.6)
+      PYBIN="/opt/python/cp36-cp36m/bin/python"
+      ;;
+    3.7)
+      PYBIN="/opt/python/cp37-cp37m/bin/python"
+      ;;
+    3.8)
+      PYBIN="/opt/python/cp38-cp38/bin/python"
+      ;;
+    3.9)
+      PYBIN="/opt/python/cp39-cp39/bin/python"
+      ;;
+    esac
+
+    yum install ninja-build cmake mesa-libGL-devel mesa-libGLU-devel -y
+
+    rm -rf /io/build
+    mkdir /io/build -p
+    cd /io/build
+
+    cmake -GNinja \
+          -DCMAKE_BUILD_TYPE=Release \
+          -DVTK_BUILD_TESTING=OFF \
+          -DVTK_BUILD_DOCUMENTATION=OFF \
+          -DVTK_BUILD_EXAMPLES=OFF \
+          -DVTK_DATA_EXCLUDE_FROM_ALL:BOOL=ON \
+          -DVTK_MODULE_ENABLE_VTK_PythonInterpreter:STRING=NO \
+          -DVTK_WHEEL_BUILD=ON \
+          -DVTK_PYTHON_VERSION=3 \
+          -DVTK_WRAP_PYTHON=ON \
+          -DVTK_OPENGL_HAS_EGL=False \
+          -DPython3_EXECUTABLE=$PYBIN ../
+    ninja-build
+
+    # build wheel
+    rm -rf dist
+    $PYBIN setup.py bdist_wheel
+
+    # cleanup wheel
+    rm -rf wheelhouse
+    auditwheel repair dist/*.whl
+    cp wheelhouse/vtk*.whl /io/wheels
+
+This script can be called with:
+
+.. code::
+
+    PYTHON_VERSION=3.8
+    docker run -e \
+           --rm -v `pwd`:/io quay.io/pypa/manylinux2014_x86_64 \
+           /io/build_wheels.sh $PYTHON_VERSION
+
+.. note::
+   To build the EGL version of the wheel, follow the directions in the
+   previous section.  Add ``mesa-libEGL-devel`` to the installation
+   dependencies.
+
+
+Python VTK Wheel on Raspberry PI (64-bit)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+While it's possible to build on 32-bit Raspberry Pi (ARMv7), there's
+several issues that crop up when building wheels for the 32-bit
+version (see `manylinux issue 84
+<https://github.com/pypa/manylinux/issues/84>`_.  Should you attempt
+to build on 32-bit , attempt to build the wheel using `dockcross
+<https://github.com/dockcross/dockcross>`_ as you may run into memory limitations otherwise (especially with only 1 GB RAM).
+
+Building the ``aarch64`` manylinux wheel can be done via docker with the ``quay.io/pypa/manylinux2014_aarch64`` image.  Run the following:
+
+.. code::
+
+    PYTHON_VERSION=3.6
+    rm -rf build
+    docker run -e \
+           --rm -v `pwd`:/io quay.io/pypa/manylinux2014_aarch64 \
+           /io/build_wheels.sh $PYTHON_VERSION
+
+Where ``build_wheels.sh`` is:
+
+.. code::
+
+
+    #!/bin/bash
+    # builds python wheels on docker container and tests installation
+
+    set -e -x
+
+    # build based on python version from args
+    PYTHON_VERSION="$1"
```

Comment:
I noticed that the RPi build_wheels.sh has EGL by default. Do you expect the typical use case on RPi to be a headless server? Or is there another reason for this? I don't own a RPi and I don't have a use case for them, so I can't judge.

Reply:
Agreed.  I think default builds should be without EGL.